### PR TITLE
Copy abi

### PR DESCRIPTION
**Pull request by KillariDev**
Allow user to input custom ABI by pasting it

image

### CODE REVIEW

File: app/ts/components/pages/AddNewAddress.tsx [hunk 1]
```diff
@@ -111,15 +94,33 @@ export const CellElement = (param: { element: ComponentChildren }) => {
...
+
+function AbiInput({ abiInput, setAbiInput, disabled }: AbiInputParams) {
+	const ref = createRef<HTMLInputElement>()
+    useEffect(() => { ref.current && ref.current.focus() }, [])
+	return <input
+		className = 'input is-spaced'
+		type = 'text'
+		value = { abiInput }
+		placeholder = { 'not available / not retrieved' }
+		onInput = { e => setAbiInput((e.target as HTMLInputElement).value) }
```
Comment (jubalm) [inline app/ts/components/pages/AddNewAddress.tsx:111]:
```suggestion
		onInput = { e => setAbiInput(e.currentTarget.value) }
```

File: app/ts/components/subcomponents/PasteCatcher.tsx [hunk 2]
```diff
@@ -7,20 +7,15 @@ interface PasteProps {
...
 	useEffect(() => {
 		window.addEventListener('paste', catcher)
-		return () => {
-			window.removeEventListener('paste', catcher);
-		}
+		return () => { window.removeEventListener('paste', catcher) }
 	}, [props.enabled, props.onPaste])
-
-	return <div></div>;
+	return <div></div>
```
Comment (jubalm) [inline app/ts/components/subcomponents/PasteCatcher.tsx:20]: are the blank `div` pairs needed here?

```suggestion
	return <></>
```

File: app/ts/simulation/services/EtherScanAbiFetcher.ts [hunk 3]
```diff
@@ -1,32 +1,55 @@
+import { ethers } from 'ethers'
 import { EtherscanGetABIResult, EtherscanSourceCodeResult } from '../../types/etherscan.js'
 import { EthereumAddress } from '../../types/wire-types.js'
-import { addressString } from '../../utils/bigint.js'
+import { addressString, checksummedAddress } from '../../utils/bigint.js'
 
-const EtherScanABIKey = 'PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8'
+const EtherscanABIKey = 'PSW8C433Q667DVEX5BCRMGNAH9FSGFZ7Q8'
```
Comment (jubalm) [inline app/ts/simulation/services/EtherScanAbiFetcher.ts:6]: would be a good candidate to move to `constants.ts`

**Review by jubalm — APPROVED** [review]

File: app/ts/simulation/services/EtherScanAbiFetcher.ts (hunk 3 above)
Comment (KillariDev) [reply app/ts/simulation/services/EtherScanAbiFetcher.ts:6]: I intended to move it for user to set, but yeah, it also should be in constant. I created issue about it

**Pull request merged**